packages/flutter/lib/src/widgets/_window_linux.dart

```diff
@@ -724,3 +770,163 @@ class RegularWindowControllerLinux extends RegularWindowController {
+  /// has a view associated with it.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  ///
+  /// See also:
+  ///
+  ///  * [DialogWindowController], the base class for dialog windows.
+  @internal
+  DialogWindowControllerLinux({
+    required WindowingOwnerLinux owner,
+    required DialogWindowControllerDelegate delegate,
+    Size? preferredSize,
+    BoxConstraints? preferredConstraints,
+    BaseWindowController? parent,
+    String? title,
+  }) : _owner = owner,
+       _delegate = delegate,
+       _parent = parent,
+       _window = _GtkWindow(),
+       super.empty() {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_kWindowingDisabledErrorMessage);
+    }
+
+    _window.setTypeHint(_GDK_WINDOW_TYPE_HINT_DIALOG);
+    if (parent != null) {
+      final controller = owner._regularWindowControllers[parent.rootView.viewId];
+      if (controller != null) {
+        _window.setTransientFor(controller._window);
+        _window.setModal(true);
+      }
+    }
+
+    _windowMonitor = _FlWindowMonitor(
+      _window,
+      // onConfigure
+      notifyListeners,
+      // onStateChanged
+      notifyListeners,
+      // onIsActiveNotify
+      notifyListeners,
+      // onTitleNotify
+      notifyListeners,
+      // onClose
+      () {
+        _delegate.onWindowCloseRequested(this);
+      },
+      // onDestroy
+      _delegate.onWindowDestroyed,
+    );
+    if (preferredSize != null) {
+      _window.setDefaultSize(preferredSize.width.toInt(), preferredSize.height.toInt());
+    }
+    if (preferredConstraints != null) {
+      setConstraints(preferredConstraints);
+    }
+    if (title != null) {
+      setTitle(title);
+    }
+    final _FlView view = _FlView();
+    final int viewId = view.getId();
+    rootView = WidgetsBinding.instance.platformDispatcher.views.firstWhere(
+      (FlutterView view) => view.viewId == viewId,
+    );
+    view.show();
+    _window.add(view);
+    _window.present();
+  }
+
+  final WindowingOwnerLinux _owner;
+  final DialogWindowControllerDelegate _delegate;
+  final _GtkWindow _window;
+  final BaseWindowController? _parent;
+  late final _FlWindowMonitor _windowMonitor;
+  bool _destroyed = false;
+
+  @override
+  @internal
+  Size get contentSize => _window.getSize();
+
+  @override
+  void destroy() {
+    if (_destroyed) {
+      return;
+    }
+    _window.destroy();
+    _windowMonitor.close();
+    _windowMonitor.unref();
+    _destroyed = true;
+    _owner._dialogWindowControllers.remove(rootView.viewId);
+  }
+
+  @override
+  @internal
+  BaseWindowController? get parent => _parent;
+
+  @override
+  @internal
+  String get title => _window.getTitle();
+
+  @override
+  @internal
+  bool get isActivated => _window.isActive();
+
+  @override
+  @internal
+  // NOTE: On Wayland this is never set, see https://gitlab.gnome.org/GNOME/gtk/-/issues/67
+  bool get isMinimized => (_window.getWindow().getState() & _GDK_WINDOW_STATE_ICONIFIED) != 0;
```

Comment:
I think the minimize API is wrong, see https://github.com/flutter/flutter/pull/177816